pkg/crc/api/types.go

```diff
@@ -11,13 +11,15 @@ import (
 
 type handlerFunc func(machine.Client, config.Storage, json.RawMessage) string
 
+type newViperConfigFunc func() (*config.Config, *config.ViperStorage, error)
```

Comment:
surprised to see concrete types here rather than interfaces? Would be nicer as:
```diff
-type newViperConfigFunc func() (*config.Config, *config.ViperStorage, error)
+type newConfigFunc func() (config.Storage, error)
```